### PR TITLE
Add testing infrastructure

### PR DESCRIPTION
**Pull request by kylef**
I just wanted to add a testing infrastructure so that it's easy to write tests for any new code added. Not focussed on adding tests to all the existing code. This includes configuring travis.

I've also added rubocop to ensure consistent syntax.

/cc @fosrias

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # HypermediaResource
 
+[![Build Status](http://img.shields.io/travis/the-hypermedia-project/hypermedia-resource-ruby/master.svg?style=flat)](https://travis-ci.org/the-hypermedia-project/hypermedia-resource-ruby)
```
Comment (fosrias) [inline README.md]: @kylef Please update this to point to /representor-ruby as I have renamed the repo already.

File: Rakefile [hunk 2]
```diff
@@ -0,0 +1,24 @@
+require 'bundler/setup'
```
Comment (fosrias) [inline Rakefile:1]: Won't block on this, but prefer from the start to have rakefile just setup rake and then load tasks in a /tasks folder for the project. Let me know what you want to do on this @kylef

Comment (kylef) [inline Rakefile:1]: I've never seen anyone do this, do you know of any examples I can take a look at?

Comment (fosrias) [inline Rakefile:1]: @kylef I will refactor and you can see the example.

**Pull request merged**